RF24Mesh.cpp

```diff
@@ -88,13 +86,91 @@ uint8_t RF24Mesh::update(){
   }
     
 	#endif
+
+	#ifndef MESH_NOMASTER
+	//Save the lastTimeSeen info each time mesh.update is called on MASTER. No action is required if this is a node
+	if(!mesh_address && type>0){//this is master and a message has been received
+		uint16_t *currAddr = (uint16_t*)network.frame_buffer;
+		int16_t currNodeID=getNodeID(*currAddr); //to access the value use *currAddr!
+		updateNodeTime(currNodeID);
+	}
+	#endif
+
 	return type;
 }
 
+/////////////////////////////////////////////////////////////
+//WARNING: _nodeID is a global variable! do not use as input!
+/////////////////////////////////////////////////////////////
+
+#ifndef MESH_NOMASTER
+bool RF24Mesh::updateNodeTime(int16_t nodeID){//this is a private function
+	bool res=false;
+	if(nodeID>0){
+		for(uint8_t i=0; i<addrListTop; i++){
+			if(addrList[i].nodeID==nodeID){
+				uint32_t _lastTime=millis();
+				addrList[i].lastTime=_lastTime;
+				res=true;
+				//Serial.println(F(" nodeID updated time"));
+				break;
+			}
+		}
+	}
+	return res;
+}
+
+/*****************************************************/
+bool RF24Mesh::remove(int16_t nodeID){
+	bool res=false;
+	//only the master can perform this action
+	if(!mesh_address && nodeID>0){
+		//The element must be put at the end of array
+		for(uint8_t i=0; i<addrListTop; i++){
+			if(addrList[i].nodeID==nodeID){
+				//Save the last (good) element at position
+				addrList[i]=addrList[addrListTop-1];
+				//Now delete the last (bad) element
+				--addrListTop;
+				addrList = (addrListStruct*)realloc(addrList,(addrListTop) * sizeof(addrListStruct));
+				res=true;
+				//Serial.print(nodeID);
+				//Serial.println(F(" nodeID removed"));
+				break;
+			}
+		}
+	}
+	return res;
+}
+
+void RF24Mesh::removeDeadNodes(uint32_t maxTime){
+	if(!mesh_address){
+		for(uint8_t i=0; i<addrListTop; i++){
+			uint32_t lastTime=millis();
+			uint32_t nodeTime=addrList[i].lastTime;
+			if(nodeTime>lastTime){//it means millis() restarted from 0
+				addrList[i].lastTime=lastTime;
+			} else{
+				if(lastTime-nodeTime>maxTime) remove(addrList[i].nodeID);
+			}
+		}
+	}
+}
+#endif
+
 bool RF24Mesh::write(uint16_t to_node, const void* data, uint8_t msg_type, size_t size ){
     if(mesh_address == MESH_DEFAULT_ADDRESS){ return 0; }
+	
+	bool _success=false;
 	RF24NetworkHeader header(to_node,msg_type);	
-	return network.write(header,data,size);	
+	//return network.write(header,data,size);	
+	uint32_t ltime = millis();
```

Comment:
Though this part still gets compiled and included. Maybe "#ifdef MESH_WRITE_RETRY"? It also makes this function blocking longer until sent.